### PR TITLE
HW-2

### PR DESCRIPTION
**Pull request by yulyaolshanska**
1. pattern examples
2. fixed code after mentor review

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    window.scrollTo({ top: 0, behavior: "smooth" });
+  };
+
+  const handleNextClick = () => {
+    setCurrentPage(currentPage + 1);
+    onPaginationClick(currentPage + 1);
+    window.scrollTo({ top: 0, behavior: "smooth" });
+  };
+
+  const isLastPage = currentPage === totalPageCount ? true : false;
```
Comment (v4voloshyn) [inline src/components/Pagination/Pagination.tsx:31]: You can reduce code to ```const isLastPage = currentPage === totalPageCount```

File: src/pages/coursesPage/CoursesPage.tsx [hunk 2]
```diff
@@ -21,7 +22,11 @@ const CoursesPage: React.FC = () => {
 
   return (
     <>
-      <CoursesList isLoading={isLoading} courses={courses} />
+      {!isLoading ? (
+        <CoursesList isLoading={isLoading} courses={courses} />
```
Comment (v4voloshyn) [inline src/pages/coursesPage/CoursesPage.tsx:26]: `isLoading={isLoading}` prop isLoading always will be false. In another case this component just will not be rendered. I think this prop is useless, or maybe I didn't catch the point.

File: src/redux/auth/authReducers.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { PayloadAction } from "@reduxjs/toolkit";
+
+export const setTokenReducer = (
+  state: { token: string | null },
```
Comment (v4voloshyn) [inline src/redux/auth/authReducers.ts:4]: `{ token: string | null }`  This code is repeated in this file, so it's better to put it in a separate interface above.

**Review by v4voloshyn — APPROVED** [review]
Need small issues fix. But the other code looks good.

File: src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (yulyaolshanska) [reply src/components/Pagination/Pagination.tsx:31]: done

**Review by v4voloshyn — APPROVED** [review]
Looks good to me :)

**Comment by artemsukach** [comment]
Looks very good!  I liked the pattern examples.

**Review by artemsukach — APPROVED** [review]

**Pull request merged**